### PR TITLE
Prevent sleep on iOS while the local web server is running or ScummVM is downloading files

### PR DESCRIPTION
**Pull request by criezy**
This pull request implements the following:
1. Add new `taskStarted` and `taskFinished` in `OSystem` to notify the backend when starting of finishing a task.
2. Call this new function in the following cases:
    1. Running the local web server
    2. Downloading games or savegames from the cloud.
    3. Download icons and shaders pack.
3. Implement this in the iOS backend to prevent sleep while one of those task is running.

I was also trying to use this to keep the network tasks running in background, which is required for example for the cloud quick connection wizard (as it starts the local web server, and then opens safari which needs to connect to the local web server now in background). But I cannot seem to get this working, so I did not include the code in this pull request.

Also I need to do more checks to make sure the calls to `taskStarted` and `taskFinished` are correctly balanced for all cases.

### CODE REVIEW

**Comment by criezy** [comment]
I have now tested that icons download, game download from cloud and local web server all temporarily disable sleep and then restore it when finished (there was an issue for the cloud download but I force-pushed a fix).

So if this approach looks good to others, I think this can now be merged. The issue of keeping the network tasks running in background can be looked at separately.

**Comment by bluegr** [comment]
Nice idea, and I like the implementation!

File: common/system.h [hunk 1]
```diff
@@ -326,6 +326,36 @@ class OSystem : Common::NonCopyable {
 	 */
 	virtual void engineDone() { }
 
+	/**
+	 * Ixdentify a task that ScummVM can perform.
```
Comment (larsamannen) [inline common/system.h:330]: Spelling on Identify

File: common/system.h [hunk 2]
```diff
@@ -326,6 +326,36 @@ class OSystem : Common::NonCopyable {
...
+	 */
+	enum Task {
+		/**
+		 * The local server is running, allowing connections from other devices to transfer files.
+		 */
+		kLocalServer,
+
+		/**
+		 * ScummVM is downloading games or sync'ing savegames from the cloud.
```
Comment (larsamannen) [inline common/system.h:339]: Better type out synchronising

File: common/system.h [hunk 3]
```diff
@@ -326,6 +326,36 @@ class OSystem : Common::NonCopyable {
...
+		 */
+		kCloudDownload,
```
Comment (larsamannen) [inline common/system.h:341]: Do we need different tasks for CloudDownload and DataPackDownload? Or could we unify them into kRemoteServer meaning a ScummVM connection to a remote server?

File: backends/cloud/storage.cpp [hunk 4]
```diff
@@ -72,8 +74,10 @@ void Storage::requestFinishedCallback(Networking::Request *invalidRequestPointer
 	--_runningRequestsCount;
 	if (_syncRestartRequestsed)
 		restartSync = true;
-	if (_runningRequestsCount == 0 && !restartSync)
+	if (_runningRequestsCount == 0) {
```
Comment (larsamannen) [inline backends/cloud/storage.cpp:77]: restartSync is removed as condition, why?

File: common/system.h (hunk 3 above)
Comment (criezy) [reply common/system.h:341]: I could also remove the Task enum entirely as I don't currently need that information. But I though that in the future backends may need it. The current granularity is based on the implementation. I for example initially intended to separate `kCloudSavegameSync` and `kCloudGameDownload`. But it was easier in the cloud code to group the two. On the other end the data pack download uses a completely different code so is trivial to separate. But I could indeed easily group it with the CloudDownload in this enum if the separation is deemed useless.

File: backends/cloud/storage.cpp (hunk 4 above)
Comment (criezy) [reply backends/cloud/storage.cpp:77]: To make sure the `taskStarted` and `taskFinished` calls are balanced.
If `restartSync` is true, this calls `syncSaves()` a few lines below, which in turn will call `addRequest` and then `taskStarted` if `_runningRequestsCount` is true. So if I had kept the old check we would have two consecutive calls to `taskStaretd` with a single `taskFinished` at the end.

I decided that for the debug message it did not really matter, so I change the test for it as well to keep the code simpler.

**Comment by larsamannen** [comment]
LGTM

**Comment by sev-** [comment]
Great feature! And I have no notes. I like the granularity. Merging

**Pull request merged**